src/App.jsx

```diff
@@ -1,12 +1,54 @@
-import { useState } from 'react'
+import { useEffect, useState } from "react";
 import './App.css'
+import Header from '../src/components/Header'
+import ArticlesList from './components/ArticlesList'
+import { Route, Routes } from "react-router-dom";
+import { getArticles } from "./utils";
+import { getTopics } from "./utils";
+
+
 
 function App() {
+  const [allArticles, setAllArticles] = useState([])
+  const [allTopics, setAllTopics] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [isError, setIsError] = useState(false)
+
+  useEffect(() => {
```

Comment:
The 'isLoading' state should be false by default, and set to true here, just before fetching. This way it gets toggled the right way every time you fetch.